src/components/headerV2/headerV2.js

```diff
@@ -1,54 +1,213 @@
 import DropdownV2 from '../toolbox/dropdownV2/dropdownV2';
+import SpinnerV2 from '../spinnerV2/spinnerV2';
+import svg from '../../utils/svgIcons';
 
 class HeaderV2 extends React.Component {
+  // eslint-disable-next-line max-statements
   constructor() {
     super();
+    const { liskCoreUrl } = getAutoLogInData();
+    let loginNetwork = findMatchingLoginNetwork();
+    let address = '';
+
+    if (loginNetwork) {
+      loginNetwork = loginNetwork.slice(-1).shift();
+    } else if (!loginNetwork) {
+      loginNetwork = liskCoreUrl ? networks.customNode : networks.default;
+      address = liskCoreUrl || '';
+    }
 
     this.state = {
+      address,
       showDropdown: false,
+      network: loginNetwork.code,
+      isFirstTime: true,
     };
 
+    this.getNetworksList();
+
     this.toggleDropdown = this.toggleDropdown.bind(this);
   }
 
-  toggleDropdown() {
-    const showDropdown = !this.state.showDropdown;
-    this.setState({ showDropdown });
+  getNetworksList() {
+    this.networks = Object.keys(networks)
+      .filter(network => network !== 'default')
+      .map((network, index) => ({
+        label: i18next.t(networks[network].name),
+        value: index,
+      }));
+  }
+
+  changeAddress({ target }) {
+    const address = target.value;
+    this.setState({
+      address,
+    });
   }
 
+  changeNetwork(network) {
+    this.setState({
+      network,
+      ...validateUrl(this.state.address),
+    });
+    this.props.settingsUpdated({ network });
+  }
+
+  getNetwork(chosenNetwork) {
+    const network = { ...getNetwork(chosenNetwork) };
+    if (chosenNetwork === networks.customNode.code) {
+      network.address = addHttp(this.state.address);
+    }
+    return network;
+  }
+
+  validateCorrectNode(network, address, nextPath) {
+    const nodeURL = address !== '' ? addHttp(address) : address;
+
+    if (network === networks.customNode.code) {
+      const liskAPIClient = new Lisk.APIClient([nodeURL], {});
+      liskAPIClient.node.getConstants()
+        .then((res) => {
+          if (res.data) {
+            this.props.liskAPIClientSet({
+              network: {
+                ...this.getNetwork(network),
+                address: nodeURL,
+              },
+            });
+
+            this.props.history.push(nextPath);
+            this.setState({ validationError: false, showDropdown: false });
+          } else {
+            throw new Error();
+          }
+        }).catch(() => {
+          this.setState({ validationError: true, showDropdown: true });
+        });
+      this.setState({ isValidationLoading: false, isFirstTime: false });
+    } else {
+      this.props.liskAPIClientSet({ network: this.getNetwork(network) });
+      this.props.history.push(nextPath);
+      this.setState({ validationError: false });
+    }
+
+    this.setState({ network });
+  }
+
+  toggleDropdown(value) {
+    this.setState({ showDropdown: value });
+  }
+
+  /* eslint-disable complexity */
```

Comment:
This eslint violation is telling you that most of the code you wrote should be a separate component `<NetworkSwitcher>` 😉 And in `Header` component it would be only `{showNetwork && <NetworkSwitcher dark={dark}>}`. NetworkSwitcher could have its own HOC and most of the methods that currently has this component.

But it's not super necessary to do this if it's not just an easy copy-paste.